torchaudio/models/wavernn.py

```diff
@@ -347,6 +355,68 @@ def forward(self, waveform: Tensor, specgram: Tensor) -> Tensor:
         # bring back channel dimension
         return x.unsqueeze(1)
 
+    @torch.jit.export
+    def infer(self, specgram: Tensor) -> Tensor:
+        r"""Inference method of WaveRNN.
+
+        This function currently only supports multinomial sampling, which assumes the
+        network is trained on crossentropy loss.
+
+        Args:
+            specgram (Tensor): The input spectrogram to the WaveRNN with shape (n_batch, n_freq, n_time).
+
+        Return:
+            waveform (Tensor): The inferred waveform with shape (n_batch, 1, n_time).
+                1 stands for a single channel.
+        """
+
+        device = specgram.device
+        dtype = specgram.dtype
+        # make it compatible with torchscript
+        n_bits = int(torch.log2(torch.ones(1) * self.n_classes))
+
+        specgram, aux = self.upsample(specgram)
+
+        output: List[Tensor] = []
+        b_size, _, seq_len = specgram.size()
+
+        h1 = torch.zeros((1, b_size, self.n_rnn), device=device, dtype=dtype)
+        h2 = torch.zeros((1, b_size, self.n_rnn), device=device, dtype=dtype)
+        x = torch.zeros((b_size, 1), device=device, dtype=dtype)
+
+        aux_split = [aux[:, self.n_aux * i: self.n_aux * (i + 1), :] for i in range(4)]
+
+        for i in range(seq_len):
+
+            m_t = specgram[:, :, i]
+
+            a1_t, a2_t, a3_t, a4_t = [a[:, :, i] for a in aux_split]
+
+            x = torch.cat([x, m_t, a1_t], dim=1)
+            x = self.fc(x)
+            _, h1 = self.rnn1(x.unsqueeze(1), h1)
+
+            x = x + h1[0]
+            inp = torch.cat([x, a2_t], dim=1)
+            _, h2 = self.rnn2(inp.unsqueeze(1), h2)
+
+            x = x + h2[0]
+            x = torch.cat([x, a3_t], dim=1)
+            x = F.relu(self.fc1(x))
+
+            x = torch.cat([x, a4_t], dim=1)
+            x = F.relu(self.fc2(x))
+
+            logits = self.fc3(x)
+
+            posterior = F.softmax(logits, dim=1)
+
+            x = torch.multinomial(posterior, 1).float()
+            x = bits_to_normalized_waveform(x, n_bits)
+            output.append(x)
+
+        return torch.stack(output).transpose(0, 1).transpose(1, 2)
```

Comment:
nit: is (1, 2, 0) the final dimension for the output?
```suggestion
        return torch.stack(output).permute(1, 2, 0)
```

Reply:
Thanks for the suggestion, I've fixed it [here](https://github.com/pytorch/audio/pull/1704/files#diff-7f4ad259da182d3840fe4c75dd0c0b7b9fe30f31503df0eff87b1e5c9378cac4R425).